Realm/RLMObjectStore.mm

```diff
@@ -678,19 +696,46 @@ void RLMDeleteObjectFromRealm(__unsafe_unretained RLMObjectBase *const object,
 
     // move last row to row we are deleting
     if (object->_row.is_attached()) {
-        object->_row.get_table()->move_last_over(object->_row.get_index());
+        RLMTrackDeletions(realm, ^{
+            object->_row.get_table()->move_last_over(object->_row.get_index());
+        });
     }
 
     // set realm to nil
     object->_realm = nil;
 }
 
+void RLMClearTable(RLMObjectSchema *objectSchema) {
+    for (auto info : objectSchema->_observedObjects) {
+        for_each(info, [&](__unsafe_unretained auto o) {
+            [o willChangeValueForKey:@"invalidated"];
+            for (RLMProperty *prop in objectSchema.properties)
```

Comment:
Link properties need to change to null before the accessor is invalidated so that observations over the key path can get cleaned up. Other property types don't strictly need to, but it's more consistent and matches the normal kvo behavior when you're observing a keypath and an intermediate property is set to nil, which is the closest analogy to an object being deleted.